dotnet/WPF/VisualLayerIntegration/VisualLayerIntegration/BarGraphUtility/Bar.cs

```diff
@@ -48,92 +46,83 @@ public float Height
                 _height = value;
                 if (Root != null)
                 {
-                    rectGeometry.Size = new Vector2(value, Width);
-                    rectOutlineGeometry.Size = new Vector2(value, Width);
+                    _rectGeometry.Size = new Vector2(value, Width);
+                    _rectOutlineGeometry.Size = new Vector2(value, Width);
                 }
             }
         }
-        public float Width { get; set; }
-        public float Value { get; set; }
+
+        public float Width { get; private set; }
+        public double Value { get; set; }
```

Comment:
Value and Label have public setters but are also set in the constructor. If you set them later what ensures that the label and value actually change? It seems like you really only want them to be set in one place and they can never change.

Reply:
The idea is to update them as the user clicks on a new data grid item -> bar values/info updates. They aren't referenced right now but in a later update having the data there would make it easier

Reply:
Why not make them private set until you make that update? Otherwise it looks like a bug.

---
In reply to: [274674777](https://github.com/Microsoft/Windows.UI.Composition-Win32-Samples/pull/1#discussion_r274674777) [](ancestors = 274674777)